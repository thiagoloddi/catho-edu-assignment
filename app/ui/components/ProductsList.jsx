import React from 'react';
import injectSheet from 'react-jss';
import propTypes from 'prop-types';

import { formatCurrency } from '../../utils/numbers';

import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import NumberInput from '../../ui/atoms/NumberInput';

const ProductsList = ({ products, classes }) => {
	const renderBody = () => {
		return products.map(({ name, _id, product_id, price, quantity, onInputClick, onInputChange, onInputBlur }) => (
			<TableRow id={_id} className={classes.row} key={_id}>
        <TableCell>{product_id}</TableCell>
        <TableCell>{name}</TableCell>
        <TableCell align="center"><NumberInput id={_id} value={quantity} onClick={onInputClick} onChange={onInputChange} onBlur={onInputBlur} /></TableCell>
        <TableCell align="center">{formatCurrency(price / 100)}</TableCell>
      </TableRow>
    ));
  }
	
  return (
		<div>
      <Paper>
        <Table className={classes.table}>
          <TableHead>
            <TableRow>
              <TableCell>ID</TableCell>
              <TableCell >NAME</TableCell>
              <TableCell align="center">QUANTITY</TableCell>
              <TableCell align="center">FULL PRICE</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {renderBody()}
          </TableBody>
        </Table>
      </Paper>
    </div>
  );
};

ProductsList.propTypes = {
	products: propTypes.arrayOf(propTypes.shape({
		name: propTypes.string.isRequired,
		_id: propTypes.string.isRequired,
		product_id: propTypes.string.isRequired,
		price: propTypes.number.isRequired,
		quantity: propTypes.number.isRequired,
		onInputClick: propTypes.func.isRequired,
		onInputChange: propTypes.func.isRequired,
		onInputBlur: propTypes.func.isRequired
	})).isRequired,
};

ProductsList.defaultProps = {
	products: []
};

const styles = {
	table: {
		marginTop: '20px',
		'& td, & th': {
			fontSize: '14px'
		}
	},
	row: {
		'&:hover': {
			backgroundColor: '#f5f5f5',
		}
	}
};

export default injectSheet(styles)(ProductsList);