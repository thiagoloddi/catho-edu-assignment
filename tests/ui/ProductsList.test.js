import React from 'react';
import renderer from 'react-test-renderer';
import ProductsList from '../../app/ui/components/ProductsList';

const func = () => {};

it('renders correctly', () => {
  const tree = renderer
    .create(<ProductsList onRowClick={func} products={[
			{
				name: 'product',
				_id: '1',
				product_id: 'classic',
				price: 100,
				quantity: 1,
				onInputClick: func,
				onInputChange: func,
				onInputBlur: func
			}
		]} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});