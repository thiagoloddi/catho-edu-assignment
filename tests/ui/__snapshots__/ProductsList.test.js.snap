// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2"
  >
    <table
      className="MuiTable-root-28 ProductsList-table-0-1-1"
    >
      <thead
        className="MuiTableHead-root-29"
      >
        <tr
          className="MuiTableRow-root-30 MuiTableRow-head-33"
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root-35 MuiTableCell-head-36"
            scope="col"
          >
            ID
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root-35 MuiTableCell-head-36"
            scope="col"
          >
            NAME
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root-35 MuiTableCell-head-36 MuiTableCell-alignCenter-44"
            scope="col"
          >
            QUANTITY
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root-35 MuiTableCell-head-36 MuiTableCell-alignCenter-44"
            scope="col"
          >
            FULL PRICE
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root-47"
      >
        <tr
          className="MuiTableRow-root-30 ProductsList-row-0-1-2"
          id="1"
          onClick={[Function]}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root-35 MuiTableCell-body-37"
          >
            classic
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root-35 MuiTableCell-body-37"
          >
            product
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root-35 MuiTableCell-body-37 MuiTableCell-alignCenter-44"
          >
            <div
              className="NumberInput-container-0-1-3"
            >
              <img
                className="NumberInput-img-0-1-4"
                id="minus"
                onClick={[Function]}
                src="/assets/minus.svg"
              />
              <input
                className="NumberInput-input-0-1-5"
                id="1"
                onBlur={[Function]}
                onChange={[Function]}
                value={1}
              />
              <img
                className="NumberInput-img-0-1-4"
                id="plus"
                onClick={[Function]}
                src="/assets/plus.svg"
              />
            </div>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root-35 MuiTableCell-body-37 MuiTableCell-alignCenter-44"
          >
            $1.00
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
